src/posix/platform/netif.cpp

```diff
@@ -269,7 +588,22 @@ static void processTransmit(otInstance *aInstance)
     message = otIp6NewMessage(aInstance, NULL);
     VerifyOrExit(message != NULL, error = OT_ERROR_NO_BUFS);
 
-    SuccessOrExit(error = otMessageAppend(message, packet, static_cast<uint16_t>(rval)));
+#if defined(__APPLE__) || defined(__NetBSD__) || defined(__FreeBSD__)
+    if ((rval >= 4) && (packet[0] == 0) && (packet[1] == 0))
+    {
+        rval -= 4;
+        offset = 4;
+    }
+#endif
+
+    if (gLogPackets)
+    {
+        fprintf(stderr, "Packet to NCP (%ld bytes)\r\n", (long)rval);
```

Comment:
Use `otLog` rather than stderr. (Applies to other `fprintf` as well).

Reply:
Done.